frontend/src/app/main/data/workspace/libraries.cljs

```diff
@@ -716,6 +756,48 @@
                   :callback do-dismiss}]
                 :sync-dialog))))))
 
+(defn watch-component-changes
+  "Watch the state for changes that affect to any main instance. If a change is detected will throw
+  an update-component-sync, so changes are immediately propagated to the component and copies."
+  []
+  (ptk/reify ::watch-component-changes
+    ptk/WatchEvent
+    (watch [_ state stream]
+      (let [components-v2 (features/active-feature? state :components-v2)
+
+            stopper
+            (->> stream
+                 (rx/filter #(or (= :app.main.data.workspace/finalize-page (ptk/type %))
+                                 (= ::watch-component-changes (ptk/type %)))))
+
+            workspace-data-str
+            (->> (rx/concat
+                  (rx/of nil)
+                  (rx/from-atom refs/workspace-data {:emit-current-value? true})))
+
+            change-str
```

Comment:
I think use of str suffix is confusing, `str` can mean `string`, I think `-s` is a better and shorter one

Reply:
Don't like very much the suffix -s, but you are true that -str is normally used for strings. Can't think of one better, so I've changed it to -s.